### ISSUE TITLE
 Add selenium tests for Slider and RangeSlider

### ISSUE THREAD

**Issue by zoohair** [bokeh/bokeh#6516]
Any chance to have that fixed or is there a workaround?

```
from bokeh.models import TextInput, RangeSlider
from bokeh.layouts import widgetbox
from bokeh.io import curdoc
from bokeh.client import push_session

def on_slider_change(attr, old, new):
    text.value = str(new[1])

def on_text_change(attr, old, new):
    try:
        n = int(new)
        n = max(n, -10)
        n = min(n, 10)

        slider.range = (0, n)

    except ValueError:
        return

text = TextInput(
        value=str(2),
        title='Population Beta:')
text.on_change('value', on_text_change)

slider = RangeSlider(
        start=0,
        end=10,
        range=(0, 2),
        step=1,
        title='Population Beta')

slider.on_change('range', on_slider_change)

curdoc().add_root(widgetbox(text, slider))

#session = push_session(curdoc())
#session.show(widgetbox(text, slider)) # open the document in a browser
#session.loop_until_closed() # run forever
```

**Comment by zoohair** [comment]
Also looks like the disabled attribute is ignored? (i.e. setting disabled = True does not disable widget)

**Comment by bryevdv** [comment]
There is a separate issue to implement `disabled` for all widgets. 

Regarding the OP, looking at the source code, it does not appear the proper change handlers are preset to update when `range` changes:

https://github.com/bokeh/bokeh/blob/master/bokehjs/src/coffee/models/widgets/range_slider.coffee

Probably a trivial fix for the next release, but the only workaround before then (or before a dev build that has a fix) would be to implement a custom extension. 

ping @mattpap

**Comment by bryevdv** [comment]
Oh, actually this capability was specifically lost in https://github.com/bokeh/bokeh/pull/6379 which fixed a catastrophic rendering problem making the slider unusable in all cases. Restoring the one line to update when `range` changes brings back the visual bug. 

I am not sure what a real solution for all cases iss offhand, will take an investigation.

**Comment by bryevdv** [comment]
I think the problem is that instead of calling `render` we cant to call whatever method the underlying slider has to update. Given that we are replacing all jq-ui including sliders, will wait to look at this until after that work is merged.

**Comment by zoohair** [comment]
Thanks @bryevdv : any hints on how to implement a custom extension in the meantime?

**Comment by bryevdv** [comment]
Well the work to remove jq and the old slider was just merged so we should probably first confirm whether this is still an issue at all on master. It's going to be a few days at lest before I can look at anything and the next few weeks are pretty rough in general. If you need to implement an extension, all the docs there are (mostly heavily annotated examples) are here: http://bokeh.pydata.org/en/latest/docs/user_guide/extensions.html

**Comment by bryevdv** [comment]
@zoohair this code seems to work as expected for me:
```
from bokeh.models import TextInput, RangeSlider
from bokeh.layouts import widgetbox
from bokeh.io import curdoc
from bokeh.client import push_session

def on_slider_change(attr, old, new):
    text.value = str(new[1])

def on_text_change(attr, old, new):
    try:
        n = int(new)
        n = max(n, -10)
        n = min(n, 10)

        slider.value = (slider.value[0], n)

    except ValueError:
        return

text = TextInput(
        value=str(2),
        title='Population Beta:')
text.on_change('value', on_text_change)

slider = RangeSlider(
        start=0,
        end=10,
        value=(0, 2),
        step=1,
        title='Population Beta')

slider.on_change('value', on_slider_change)

curdoc().add_root(widgetbox(text, slider))

```

@mattpap it looks like the new slider has `value` instead of `range`? I'm OK with that but we need  a migration note

**Comment by bryevdv** [comment]
OK the original issue was resolved (I just checked again with 1.1.1dev1) but we should add an integration test now that is is easier to do
